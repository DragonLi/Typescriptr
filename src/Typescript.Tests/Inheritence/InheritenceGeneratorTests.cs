using System;
using Assent;
using Typescriptr;
using Typescriptr.Formatters;
using Xunit;
using Xunit.Abstractions;

namespace Typescript.Tests.Inheritence
{
    public class InheritenceGeneratorTests
    {
        class BaseClass
        {
            public string Property { get; set; }
        }

        class TypeWithBaseClass : BaseClass
        {
            
        }

        [Fact]
        public void Generator_TypeWithBaseClass_GeneratesSuccessfully()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] {typeof(TypeWithBaseClass) });

            this.Assent(generated.Types);
        }


        class TypeWithGenericParent : GenericParent<TypeWithGenericParent>
        {

        }
        class GenericParent<T>
        {
        }

        [Fact]
        public void Generator_TypeWithGenericParent_ShouldRenderValidTypescript()
        {
            var generator = TypeScriptGenerator.CreateDefault();
            var generated = generator.Generate(new[] { typeof(TypeWithGenericParent) });

            this.Assent(generated.Types);
        }
    }
}