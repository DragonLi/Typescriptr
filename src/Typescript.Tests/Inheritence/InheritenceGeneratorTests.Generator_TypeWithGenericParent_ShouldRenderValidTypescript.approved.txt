declare namespace Api {
  interface TypeWithGenericParent {
  }
}
